### PR TITLE
Adding security limitations for PSI-Cardinality Protocol

### PR DESCRIPTION
**Pull request by rutujasurve94**

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+1. Server set size
+2. Client set size
+(Note that Each of these can be turned into upper bounds by adding dummy elements.)
+
+### Security Limitations for the PSI-Cardinality protocol
+
+Coordinated clients could get the actual intersection. However, server set items not
+in any of the client sets will never be uncovered.
+Situations where it’s feasible for clients to send one request per element in the domain -
+there is a possbility that oordinated clients could uncover server set. A larger set means
```
Comment (s0l0ist) [inline SECURITY.md:14]: *coordinated

File: SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+your health authority only covers 10 possible geohashes, people could sidestep the PSI step
+entirely and submit location histories which unlock tests by brute force.
+
+
+A potential limitation with the PSI-cardinality approach is the communication complexity,
+which scales linearly with the size of the larger set. This is of particular concern
+when performing PSI between a constrained device (cellphone) holding a small set, and a
+large service provider (e.g. WhatsApp), such as in the Private Contact Discovery application.
+Our protocol has communication complexity linear in the size of the smaller set, and
+logarithmic in the larger set.
```
Comment (s0l0ist) [inline SECURITY.md:38]: Could we add a bit more clarity on who is holding the smaller/larger set?

Ex:
The Client set size affects the algorithmic complexity in linear time O(n).
The Client set size affects the algorithmic complexity in logarithmic time O(log(n)).

Comment (schoppmp) [reply SECURITY.md:38]: It's actually linear in both. The bloom filter has linear size in the server's set.

Comment (rutujasurve94) [reply SECURITY.md]: Made changes as per this to the PR.

File: SECURITY.md (hunk 1 above)
Comment (rutujasurve94) [reply SECURITY.md]: Done

**Pull request merged**